src/Runner.Worker/ActionManager.cs

```diff
@@ -504,27 +502,83 @@ private async Task DownloadRepositoryActionAsync(IExecutionContext executionCont
                 executionContext.Output($"Download action repository '{repositoryReference.Name}@{repositoryReference.Ref}'");
             }
 
+            var configurationStore = HostContext.GetService<IConfigurationStore>();
+            var isHostedServer = configurationStore.GetSettings().IsHostedServer;
+            if (isHostedServer)
+            {
+                string apiUrl = GetApiUrl(executionContext, forceDotCom: true);
+                string archiveLink = BuildLinkToActionArchive(apiUrl, repositoryReference.Name, repositoryReference.Ref);
+                Trace.Info($"Download archive '{archiveLink}' to '{destDirectory}'.");
+                await DownloadRepositoryActionAsync(executionContext, archiveLink, destDirectory);
+            }
+            else
+            {
+                string apiUrl = GetApiUrl(executionContext);
+
+                // URLs to try:
+                var archiveLinks = new List<string> {
+                    // API_URL/repos/{repository}/zipball/{ref}  - A built-in action or an action the user has created, on their GHES instance
+                    BuildLinkToActionArchive(apiUrl, repositoryReference.Name, repositoryReference.Ref),
+                    // API_URL/repos/actions/community-{repository.replace(/,_)}/zipball/{ref}  - A community action, synced to their GHES instance
+                    BuildLinkToActionArchive(apiUrl, $"actions/community-{repositoryReference.Name.Replace("/", "_")}", repositoryReference.Ref)
+                };
+
+                foreach (var archiveLink in archiveLinks)
+                {
+                    Trace.Info($"Download archive '{archiveLink}' to '{destDirectory}'.");
+                    try
+                    {
+                        await DownloadRepositoryActionAsync(executionContext, archiveLink, destDirectory);
+                        return;
+                    }
+                    catch (ActionNotFoundException ex)
+                    {
+                        Trace.Info($"Failed to find the action at {ex.ActionUri}");
+                        continue;
+                    }
+                }
+                throw new NotSupportedException($"Failed to find the action '{repositoryReference.Name}'.  Paths attempted: {string.Join(", ", archiveLinks)}");
+            }
+        }
+
+        private string GetApiUrl(IExecutionContext executionContext, bool forceDotCom = false)
```

Comment:
Eventually this will be set for hosted too. I would just fallback to `https://api.github.com` for back compat.

Reply:
+1, remove the `forceDotCom`, just failback.